tegel/src/components/accordion/accordion-item/accordion-item.tsx

```diff
@@ -21,19 +21,27 @@ export class AccordionItem {
   /** When true 16px on right padding instead of 64px */
   @Prop() paddingReset: boolean = false;
 
-  /** Fires after the accordion item is closed or opened, emitting the value (as boolean) of the current state of the accordion */
+  /** Method for toggeling the expanded state of the accordion item. */
+  @Method()
+  async toggleAccordionItem() {
+    const event = this.sddsToggle.emit({
+      expanded: this.expanded,
+    });
+    if (!event.defaultPrevented) {
+      this.expanded = !this.expanded;
+    }
+  }
```

Comment:
```suggestion
  async toggleAccordionItem() {
  	const toggledExpanded = !this.expanded;
    const event = this.sddsToggle.emit({
      expanded: toggledExpanded,
    });
    if (!event.defaultPrevented) {
      this.expanded = toggledExpanded;
    }
  }
```
This code emits with the expanded state before it is toggled, so it will show the old value, not the new one, which might cause confusion.

In the suggestion i made it so the event emits with the toggled value, but it is not set if default is prevented.

Reply:
Yeah I'm not sure how to handle these really. Ideally I would not want to emit a value that's "false".

The code above could just be: 
` async toggleAccordionItem() {
    const event = this.sddsToggle.emit({
      expanded: !this.expanded,
    });
    if (!event.defaultPrevented) {
      this.expanded = !this.expanded;
    }
  }`
  
  We don't really need the temp value. Would should leave a comment explaining it for future reference though

Reply:
Yea ChatGPT gave me that as an example to fix it, but then I asked to compare it to the one with the variable it gave me this 🤷 

> Both code snippets achieve the same goal of preventing the expanded state from being toggled if the event default is prevented. The code you provided is a bit more explicit, and it has some advantages:

>1. Readability: By creating a separate variable (nextExpanded) to store the opposite value of the current expanded state, the code becomes more readable and easier to understand.

>2. Maintainability: With a separate variable, it's easier to modify the code in the future if there's any change in the logic of how the next state is calculated.

>3. Debugging: Having a separate variable makes it easier to debug the code if any issues arise since you can directly inspect the nextExpanded value.

But don't think it really matters which way TBH